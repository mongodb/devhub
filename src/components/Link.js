import React from 'react';
import PropTypes from 'prop-types';
import DevHubLink from './dev-hub/link';
import { isLinkForImage } from '~utils/is-link-for-image';
import { isPreviewMode } from '~utils/is-preview-mode';

/*
 * Note: This component is not suitable for internal page navigation:
 * https://www.gatsbyjs.org/docs/gatsby-link/#recommendations-for-programmatic-in-app-navigation
 */

// Since DOM elements <a> cannot receive activeClassName and partiallyActive,
// destructure the prop here and pass it only to GatsbyLink.
const Link = React.forwardRef(
    ({ children, to, activeClassName, partiallyActive, ...other }, ref) => {
        if (!to) to = '';
        // Assume that external links begin with http:// or https:// or have mailto
        const external =
            /^http(s)?:\/\/|^mailto:[\w-.]+@(\w+.)/.test(to) ||
            isLinkForImage(to);
        const anchor = to.startsWith('#');

        // Use Gatsby Link for internal links, and <a> for others
        if (!isPreviewMode() && to && !external && !anchor) {
            if (!to.startsWith('/')) to = `/${to}`;
            return (
                <DevHubLink
                    ref={ref}
                    to={to}
                    activeClassName={activeClassName}
                    partiallyActive={partiallyActive}
                    {...other}
                >
                    {children}
                </DevHubLink>
            );
        }
        if (anchor) {
            // We don't want any target="_blank" to apply
            return (
                <DevHubLink ref={ref} href={to} {...other} target="_self">
                    {children}
                </DevHubLink>
            );
        }
        return (
            <DevHubLink ref={ref} href={to} {...other}>
                {children}
            </DevHubLink>
        );
    }
);

Link.propTypes = {
    to: PropTypes.string.isRequired,
};

export default Link;
