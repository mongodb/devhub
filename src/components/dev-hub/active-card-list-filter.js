import React from 'react';
import CardList from './card-list';
import { LearnPageTabs } from '../../utils/learn-page-tabs';

const ActiveCardListFilter = ({ activeContentTab, textFilterResults }) => {
    switch (activeContentTab) {
        case LearnPageTabs.articles:
            return <CardList articles={textFilterResults} shouldSort />;
        case LearnPageTabs.videos:
            return <CardList videos={textFilterResults} shouldSort />;
        case LearnPageTabs.podcasts:
            return <CardList podcasts={textFilterResults} shouldSort />;
        default:
            return <CardList all={textFilterResults} shouldSort />;
    }
};

export default ActiveCardListFilter;
