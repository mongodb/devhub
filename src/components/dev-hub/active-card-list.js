import React from 'react';
import CardList from './card-list';
import { LearnPageTabs } from '../../utils/learn-page-tabs';

const ActiveCardList = ({
    activeContentTab,
    articles,
    videos,
    podcasts,
    hasNoFilter,
}) => {
    switch (activeContentTab) {
        case LearnPageTabs.articles:
            return <CardList articles={articles} shouldSort />;
        case LearnPageTabs.videos:
            return <CardList videos={videos} shouldSort />;
        case LearnPageTabs.podcasts:
            return <CardList podcasts={podcasts} shouldSort />;
        default:
            return (
                <CardList
                    shouldSort
                    articles={articles}
                    videos={hasNoFilter ? videos : []}
                    podcasts={hasNoFilter ? podcasts : []}
                />
            );
    }
};

export default ActiveCardList;
