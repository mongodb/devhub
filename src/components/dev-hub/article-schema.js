import React from 'react';
import { Helmet } from 'react-helmet';

const getAuthorsList = authors =>
    authors?.map(({ name }) => ({
        '@type': 'Person',
        name,
    }));

const ArticleSchema = ({
    authors,
    articleUrl,
    description,
    imageUrl,
    modifiedDate,
    publishedDate,
    tags,
    title,
    language = 'English',
}) => (
    <Helmet>
        <script id="structured-data" type="application/ld+json">
            {JSON.stringify({
                '@context': 'https://schema.org',
                '@type': 'BlogPosting',
                url: articleUrl,
                headline: title,
                description,
                datePublished: publishedDate,
                dateModified: modifiedDate || publishedDate,
                mainEntityOfPage: {
                    '@type': 'WebPage',
                    url: articleUrl,
                },
                image: {
                    '@type': 'imageObject',
                    url: imageUrl,
                },
                publisher: {
                    '@type': 'Organization',
                    name: 'MongoDB',
                    logo: {
                        '@type': 'imageObject',
                        url: 'https://webassets.mongodb.com/_com_assets/cms/mongodb_logo1-76twgcu2dm.png',
                    },
                },
                author: getAuthorsList(authors),
                tags,
                inLanguage: language,
            })}
        </script>
    </Helmet>
);

export default ArticleSchema;
