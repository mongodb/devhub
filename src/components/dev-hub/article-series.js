import React from 'react';
import dlv from 'dlv';
import Series from './series';

const ArticleSeries = ({ allSeriesForArticle, slugTitleMapping, title }) => {
    // Handle if this article is not in a series or no series are defined
    if (!allSeriesForArticle) return null;
    const getMappedSeries = seriesSlugs => {
        if (!seriesSlugs || !seriesSlugs.length) return null;

        let hasSeenActiveSlug = false;
        const getSeriesArticlePosition = articleTitle => {
            // Find if an article is the current one in the series, or is upcoming/past
            if (hasSeenActiveSlug) {
                return 'upcoming';
            } else if (articleTitle === title) {
                hasSeenActiveSlug = true;
                return 'active';
            }
            return 'past';
        };

        const mappedSeries = seriesSlugs.map(slug => {
            const title = dlv(slugTitleMapping, [slug, 0, 'value'], slug);
            const position = getSeriesArticlePosition(title);
            return {
                position,
                slug,
                title,
            };
        });
        return mappedSeries;
    };

    return Object.keys(allSeriesForArticle).map(series => {
        const seriesItems = getMappedSeries(allSeriesForArticle[series]);
        return <Series name={series}>{seriesItems}</Series>;
    });
};

export default ArticleSeries;
