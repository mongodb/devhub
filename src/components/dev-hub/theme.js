import { css } from '@emotion/react';

/**
 * @type {Object.<string, string>}
 * @property {string} fontsize returns px value
 */
const fontSize = {
    micro: '10px',
    xsmall: '12px',
    tiny: '14px',
    small: '16px',
    default: '18px',
    medium: '20px',
    large: '24px',
    xlarge: '30px',
    xxlarge: '36px',
    jumbo: '42px',
};
/**
 * @type {Object.<string, string>}
 * @property {string} lineHeight
 */
const lineHeight = {
    micro: '16px',
    xsmall: '18px',
    tiny: '20px',
    small: '24px',
    default: '28px',
    medium: '32px',
    large: '36px',
    xlarge: '42px',
    xxlarge: '48px',
    jumbo: '54px',
};
/**
 * @type {Object}
 * @property {string} size returns px value
 */
const size = {
    tiny: '5px',
    xsmall: '8px',
    small: '10px',
    default: '16px',
    medium: '20px',
    mediumLarge: '24px',
    articleContent: '30px',
    large: '32px',
    xlarge: '64px',
    xxlarge: '120px',
    maxContentWidth: '780px',
    maxWidth: '1440px',
    /** @type {function(string): number} */
    stripUnit(unit) {
        return parseInt(unit, 10);
    },
};

const colorMap = {
    darkGreen: '#13AA52',
    devBlack: '#0C1C27',
    devWhite: '#f9fbfa',
    greyDarkOne: '#5D6C74',
    greyDarkTwo: '#3D4F58',
    greyDarkThree: '#21313C',
    greyLightOne: '#E7EEEC',
    greyLightTwo: '#B8C4C2',
    greyLightThree: '#9FA1A2',
    lightGreen: '#0AD05B',
    orange: '#F97216',
    magenta: '#D34F94',
    pageBackground: '#061621',
    salmon: '#E55F55',
    sherbet: '#F7A76F',
    teal: '#2F9FC5',
    violet: '#6E60F9',
    purple: '#4F4FBF',
    yellow: '#FFDD49',
};

const gradientMap = {
    green: `linear-gradient(
        315deg,
        ${colorMap.lightGreen} 0%,
        ${colorMap.lightGreen} 40%,
        ${colorMap.darkGreen} 100%
    )`,
    greenTeal: `linear-gradient(
        270deg,
        ${colorMap.lightGreen} 0%,
        ${colorMap.teal} 100%
    )`,
    greenTealOffset: `linear-gradient(
        315deg,
        ${colorMap.teal} 0%,
        ${colorMap.lightGreen} 100%
    )`,
    magentaSalmonSherbet: `linear-gradient(
        270deg,
        ${colorMap.sherbet} 0%,
        ${colorMap.salmon} 49.99%,
        ${colorMap.magenta} 100%
    )`,
    magentaSalmonYellow: `linear-gradient(
        270deg,
        ${colorMap.yellow} 0%,
        ${colorMap.salmon} 49.99%,
        ${colorMap.magenta} 100%
    )`,
    tealViolet: `linear-gradient(
        270deg,
        ${colorMap.teal} 0%,
        ${colorMap.violet} 100%
    )`,
    tealVioletReverse: `linear-gradient(
        270deg,
        ${colorMap.violet} 0%,
        ${colorMap.teal} 100%
    )`,
    tealVioletPurple: `linear-gradient(
        320deg, 
        ${colorMap.teal} 0%, 
        ${colorMap.violet} 33.14%, 
        ${colorMap.purple} 100%
    )`,
    violetMagenta: `linear-gradient(
        270deg,
        ${colorMap.violet} 0%,
        ${colorMap.magenta} 100%
    )`,
    violentMagentaOrange: `linear-gradient(
        270deg,
        ${colorMap.violet} 0%,
        ${colorMap.magenta} 49.99%,
        ${colorMap.orange} 100%
    )`,
    violetMagentaReverse: `linear-gradient(
        270deg,
        ${colorMap.magenta} 0%,
        ${colorMap.violet} 100%
    )`,
};

const borderGradients = {
    greenTeal: `
        border: 2px solid;
        border-image: ${gradientMap.greenTeal};
        border-image-slice: 1;
    `,
    tealViolet: `
        border: 2px solid;
        border-image: ${gradientMap.tealViolet};
        border-image-slice: 1;
    `,
    tealVioletReverse: `
        border: 2px solid;
        border-image: ${gradientMap.tealVioletReverse};
        border-image-slice: 1;
    `,
    violetMagenta: `
        border: 2px solid;
        border-image: ${gradientMap.violetMagenta};
        border-image-slice: 1;
    `,
    violetMagentaReverse: `
        border: 2px solid;
        border-image: ${gradientMap.violetMagentaReverse};
        border-image-slice: 1;
    `,
};

/**
 * store common responsive sizes
 * @type {Object.<string, string>}
 */
const screenSize = {
    upToSmall: 'only screen and (max-width: 420px)',
    smallAndUp: 'not all and (max-width: 420px)',
    upToMedium: 'only screen and (max-width: 767px)',
    mediumAndUp: 'not all and (max-width: 767px)',
    upToLarge: 'only screen and (max-width: 1023px)',
    largeAndUp: 'not all and (max-width: 1023px)',
    upToSmallDesktop: 'only screen and (max-width: 1200px)',
    smallDesktopAndUp: 'not all and (max-width: 1200px)',
    upToXlarge: `only screen and (max-width: ${size.maxWidth})`,
    xlargeAndUp: `not all and (max-width: ${size.maxWidth})`,
};

const gridLayout = {
    desktop: {
        columnGap: '24px',
        numCols: 12,
        sideMargin: size.xxlarge,
    },
    mobile: {
        columnGap: size.default,
        numCols: 4,
        sideMargin: size.default,
    },
    tablet: {
        columnGap: size.default,
        numCols: 8,
        sideMargin: size.xlarge,
    },
};

const grid = css`
    display: grid;
    grid-template-columns: repeat(${gridLayout.desktop.numCols}, 1fr);
    column-gap: ${gridLayout.desktop.columnGap};
    margin: 0 ${gridLayout.desktop.sideMargin};
    @media ${screenSize.upToLarge} {
        grid-template-columns: repeat(${gridLayout.tablet.numCols}, 1fr);
        column-gap: ${gridLayout.tablet.columnGap};
        margin: 0 ${gridLayout.tablet.sideMargin};
    }
    @media ${screenSize.upToMedium} {
        grid-template-columns: repeat(${gridLayout.mobile.numCols}, 1fr);
        column-gap: ${gridLayout.mobile.columnGap};
        margin: 0 ${gridLayout.mobile.sideMargin};
    }
`;

/**
 * z-index values in ranges of 10. This should give enough leeway to incremement in components as needed.
 * @type {Object.<string, number>}
 * @property {string} position returns a z-index value
 */
const layer = {
    superBack: -1,
    back: 1,
    middle: 11,
    front: 21,
    superFront: 22,
};

/**
 * @type {Object.<string, string>}
 * @property {string} animationSpeed
 */
const animationSpeed = {
    slow: '650ms',
    medium: '300ms',
    fast: '150ms',
};

const FORM_ELEMENT_BORDER = '2px';

const HERO_CONTENT_WIDTH = '640px';

// Use colors with emotion theme
export const darkTheme = {
    borderGradients,
    colorMap,
    gradientMap,
};

export {
    animationSpeed,
    borderGradients,
    colorMap,
    fontSize,
    FORM_ELEMENT_BORDER,
    gradientMap,
    grid,
    HERO_CONTENT_WIDTH,
    layer,
    lineHeight,
    screenSize,
    size,
};
