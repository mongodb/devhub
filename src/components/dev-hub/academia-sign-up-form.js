import React, { useCallback, useState } from 'react';
import styled from '@emotion/styled';
import { submitAcademiaForm } from '../../utils/devhub-api-stitch';
import { fontSize, size, screenSize } from './theme';
import { P, ArticleH3 } from './text';
import Input from './input';
import Button from './button';
import Select from './select';
import Checkbox from '@leafygreen-ui/checkbox';
import SectionHeader from './section-header';
import SuccessState from './success-state';

const INPUT_BOX_WIDTH = '100%';

const StyledButton = styled(Button)`
    display: flex;
    margin: ${size.large} auto;
`;

const StyledCheckbox = styled(Checkbox)`
    /* Leafygreen's Checkbox input is given height/width of 0, hiding invalid popup */
    > input {
        height: 20px;
        width: 20px;
    }
    > span {
        font-size: ${fontSize.small};
    }
`;

const StyledSelect = styled(Select)`
    color: ${({ theme }) => theme.colorMap.greyLightTwo};
`;

const ErrorMessage = styled(P)`
    color: ${({ theme }) => theme.colorMap.salmon};
`;

const StyledSectionText = styled(ArticleH3)`
    color: ${({ theme }) => theme.colorMap.greyLightTwo};
    font-family: akzidenz;
    font-weight: normal;
`;

const StyledSuccessState = styled(SuccessState)`
    margin-bottom: ${size.xlarge};
    margin-top: ${size.xlarge};
`;

const InstructorSection = styled('div')`
    display: grid;
    grid-template-columns: ${INPUT_BOX_WIDTH};
    grid-row-gap: ${size.mediumLarge};
    margin-bottom: ${size.xlarge};
    @media ${screenSize.upToMedium} {
        grid-template-columns: 100%;
    }
`;

const InstitutionSection = styled('div')`
    display: grid;
    grid-template-columns: ${INPUT_BOX_WIDTH};
    grid-row-gap: ${size.mediumLarge};
    justify-content: space-between;
    margin-bottom: ${size.xlarge};
    @media ${screenSize.upToMedium} {
        grid-template-columns: 100%;
    }
`;

const Form = React.memo(({ setSuccess, success, ...props }) => {
    const [firstName, setFirstName] = useState('');
    const [lastName, setLastName] = useState('');
    const [email, setEmail] = useState('');
    const [institutionName, setInstitutionName] = useState('');
    const [institutionType, setInstitutionType] = useState('');
    const [agreeToEmail, setAgreeToEmail] = useState(false);
    const [canSubmit, setCanSubmit] = useState(true);
    const institutionTypes = [
        ['bootcamp', 'Bootcamp'],
        ['online_course', 'Online Course'],
        ['high_school', 'High School'],
        ['university_college', 'University / College'],
    ];

    const handleSubmit = async e => {
        e.preventDefault();
        setCanSubmit(false);
        const data = {
            first_name: firstName,
            last_name: lastName,
            email,
            institution_name: institutionName,
            institution_type: institutionType,
            agree_to_email: agreeToEmail,
        };
        const response = await submitAcademiaForm(data);

        setSuccess(response.success);
        setCanSubmit(!response.success);
    };

    const onNameInvalid = (name, nameType) =>
        name === ''
            ? `${nameType} cannot be blank`
            : `${nameType} should only contain letters. e.g. John`;

    const onEmailInvalid = useCallback(
        e => {
            e.target.setCustomValidity(
                email === ''
                    ? 'Email cannot be blank'
                    : 'Please enter a valid email address. e.g. example@email.com'
            );
        },
        [email]
    );

    const onFirstNameInvalid = useCallback(
        e => {
            e.target.setCustomValidity(onNameInvalid(firstName, 'First name'));
        },
        [firstName]
    );

    const onLastNameInvalid = useCallback(
        e => {
            e.target.setCustomValidity(onNameInvalid(lastName, 'Last name'));
        },
        [lastName]
    );

    return (
        <form onSubmit={handleSubmit} {...props}>
            <ErrorMessage>
                {success === false &&
                    'Your submission failed. Please try again.'}
            </ErrorMessage>

            <SectionHeader>
                <StyledSectionText>Instructor</StyledSectionText>
            </SectionHeader>

            <InstructorSection>
                <Input
                    narrow
                    value={firstName}
                    maxLength="25"
                    required
                    pattern="^[A-Za-zÀ-ÿ ,.'-]+$"
                    placeholder="First Name"
                    onChange={e => setFirstName(e.target.value)}
                    onInput={e => e.target.setCustomValidity('')}
                    onInvalid={onFirstNameInvalid}
                />
                <Input
                    narrow
                    value={lastName}
                    maxLength="25"
                    required
                    pattern="^[A-Za-zÀ-ÿ ,.'-]+$"
                    placeholder="Last Name"
                    onChange={e => setLastName(e.target.value)}
                    onInput={e => e.target.setCustomValidity('')}
                    onInvalid={onLastNameInvalid}
                />
                <Input
                    narrow
                    type="email"
                    value={email}
                    required
                    placeholder="Email Address"
                    onChange={e => setEmail(e.target.value)}
                    onInput={e => e.target.setCustomValidity('')}
                    onInvalid={onEmailInvalid}
                />
            </InstructorSection>

            <SectionHeader>
                <StyledSectionText>Institution</StyledSectionText>
            </SectionHeader>

            <InstitutionSection>
                <Input
                    narrow
                    value={institutionName}
                    required
                    maxLength="50"
                    placeholder="Name of Institution"
                    onChange={e => setInstitutionName(e.target.value)}
                />
                <StyledSelect
                    narrow
                    value={institutionType}
                    defaultText="Institution Type"
                    choices={institutionTypes}
                    onChange={e => setInstitutionType(e)}
                />
            </InstitutionSection>

            <StyledCheckbox
                onChange={e => setAgreeToEmail(e.target.value)}
                required
                label="I agree to receive emails from MongoDB, Inc. After submitting, 
                a MongoDB representative will reach out within five business days."
                variant="light"
                title="Please submit"
            />

            <StyledButton
                disabled={!canSubmit}
                type="submit"
                primary
                hasArrow={false}
            >
                Join MongoDB for Academia
            </StyledButton>
        </form>
    );
});

const AcademiaSignUpForm = ({ ...props }) => {
    const [success, setSuccess] = useState(null);
    if (success) {
        return <StyledSuccessState>Thank you for joining!</StyledSuccessState>;
    }
    return <Form setSuccess={setSuccess} success={success} {...props} />;
};

export default AcademiaSignUpForm;
