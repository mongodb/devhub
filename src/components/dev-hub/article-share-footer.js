import React, { useCallback } from 'react';
import copy from 'copy-to-clipboard';
import styled from '@emotion/styled';
import { getArticleShareLinks } from '../../utils/get-article-share-links';
import BlogTagList from './blog-tag-list';
import { colorMap, screenSize, size } from './theme';
import Link from './link';
import Tooltip from './tooltip';
import LinkIcon from './icons/link-icon';
import LinkedIn from './icons/linkedin';
import FacebookIcon from './icons/facebook-icon';
import TwitterIcon from './icons/twitter-icon';

const ArticleShareArea = styled('div')`
    border-top: 1px solid ${colorMap.greyDarkTwo};
    display: flex;
    justify-content: space-between;
    margin-bottom: ${size.large};
    margin-top: ${size.xlarge};
    padding-top: ${size.medium};
    @media ${screenSize.upToMedium} {
        margin-top: ${size.large};
    }
`;

const BlogShareLinks = styled('div')`
    span,
    a:not(:last-of-type) {
        margin-right: ${size.medium};
    }
`;

const ArticleShareFooter = ({ tags, title, url }) => {
    const onCopyLink = useCallback(() => {
        copy(url);
    }, [url]);
    const { facebookUrl, linkedInUrl, twitterUrl } = getArticleShareLinks(
        title,
        url
    );
    return (
        <ArticleShareArea>
            <BlogTagList tags={tags} />
            <BlogShareLinks data-test="article-share-links">
                <Tooltip
                    position="bottom"
                    trigger={
                        <Link onClick={onCopyLink}>
                            <LinkIcon />
                        </Link>
                    }
                >
                    Article link copied to clipboard!
                </Tooltip>

                <Link target="_blank" href={linkedInUrl}>
                    <LinkedIn />
                </Link>

                <Link target="_blank" href={twitterUrl}>
                    <TwitterIcon />
                </Link>
                <Link target="_blank" href={facebookUrl}>
                    <FacebookIcon height="22" width="22" />
                </Link>
            </BlogShareLinks>
        </ArticleShareArea>
    );
};

export default ArticleShareFooter;
