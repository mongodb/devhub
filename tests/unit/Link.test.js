import React from 'react';
import { render } from 'enzyme';
import Link from '../../src/components/Link';
import { ThemeProvider } from 'emotion-theming';
import { darkTheme} from '../../src/components/dev-hub/theme';

const setup = ({ text, ...rest }) => render(<ThemeProvider theme={darkTheme}><Link {...rest}>{text}</Link></ThemeProvider>);

describe('Link component renders a variety of strings correctly', () => {
    it('empty string', () => {
        const tree = setup({ to: '', text: 'Empty string' });
        expect(tree).toMatchSnapshot();
    });

    it('external URL', () => {
        const tree = setup({
            to: 'http://mongodb.com',
            text: 'MongoDB Company',
        });
        expect(tree).toMatchSnapshot();
    });

    it('internal link', () => {
        const tree = setup({
            to: 'drivers/c',
            text: 'C Driver',
            className: 'test-class',
        });
        expect(tree).toMatchSnapshot();
    });
});
