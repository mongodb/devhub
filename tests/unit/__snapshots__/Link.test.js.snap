// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component renders a variety of strings correctly empty string 1`] = `
<a
  class="css-4pi1z-StyledLink eygprlc0"
  href=""
>
  Empty string
</a>
`;

exports[`Link component renders a variety of strings correctly external URL 1`] = `
<a
  class="css-4pi1z-StyledLink eygprlc0"
  href="http://mongodb.com"
>
  MongoDB Company
</a>
`;

exports[`Link component renders a variety of strings correctly internal link 1`] = `
<a
  class="test-class css-1m971tq-AsInternalLink-StyledLink eygprlc1"
  href="/drivers/c/"
>
  C Driver
</a>
`;
